homeassistant/components/ping/helpers.py

```diff
@@ -58,9 +59,16 @@ async def async_update(self) -> None:
                 timeout=ICMP_TIMEOUT,
                 privileged=self._privileged,
             )
-        except NameLookupError:
+        except NameLookupError as err:
             self.is_alive = False
-            return
+            raise UpdateFailed(f"Error resolving host: {self.ip_address}") from err
```

Comment:
It's weird to raise an exception that is specific to the update coordinator in a context that isn't aware of the coordinator.